### PR TITLE
zh-cn: translate ReadableStream's from() static method

### PR DESCRIPTION
**Pull request by fwqaaq**

### CODE REVIEW

File: files/zh-cn/web/api/readablestream/from_static/index.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
+---
+title: "ReadableStream: from() 静态方法"
+slug: Web/API/ReadableStream/from_static
+---
+
+{{APIRef("Streams")}}{{SeeCompatTable}}
+
+**`ReadableStream.from()`** 静态方法从提供的可迭代或异步迭代对象返回一个 {{domxref("ReadableStream")}}。
```
Comment (yin1999) [inline files/zh-cn/web/api/readablestream/from_static/index.md:8]:
```suggestion
**`ReadableStream.from()`** 静态方法从提供的可迭代或异步可迭代对象返回一个 {{domxref("ReadableStream")}}。
```

File: files/zh-cn/web/api/readablestream/from_static/index.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+该方法可以用于将可迭代或异步迭代对象包裹为可读流，包括数组、集合、promise 数组、异步生成器、`ReadableStream`、Node.js 可读流等等。
```
Comment (yin1999) [inline files/zh-cn/web/api/readablestream/from_static/index.md:10]:
```suggestion
该方法可以用于将可迭代或异步可迭代对象包装为可读流，包括数组、集合、promise 数组、异步生成器、`ReadableStream`、Node.js 可读流，等等。
```

File: files/zh-cn/web/api/readablestream/from_static/index.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+  - : 一个[可迭代](/zh-CN/docs/Web/JavaScript/Reference/Iteration_protocols#可迭代协议)或[异步可迭代](/zh-CN/docs/Web/JavaScript/Reference/Iteration_protocols#异步迭代器和异步可迭代协议)对象。
+
+### 返回值
+
+一个 {{domxref("ReadableStream")}}。
+
+### 异常
+
+- {{jsxref("TypeError")}}
+  - : 如果传入的参数不是可迭代对象或者异步可迭代对象（没有实现 `@@iterator` 或 `@@asyncIterator` 方法），则抛出。如果在迭代期间，下一跳的结果不是对象或者 promise 不能解析为对象，则也会抛出。
```
Comment (yin1999) [inline files/zh-cn/web/api/readablestream/from_static/index.md:30]: does not define

```suggestion
  - : 如果传入的参数不是可迭代对象或者异步可迭代对象（未定义 `@@iterator` 或 `@@asyncIterator` 方法），则抛出。如果在迭代期间，下一步的结果不是对象或者 promise 不能兑现对象，则也会抛出。
```

File: files/zh-cn/web/api/readablestream/from_static/index.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+function log(text) {
+  logElement.innerText += `${text}\n`;
+}
+```
+
+然后，检查是否支持静态方法，如果不支持，则记录不支持的结果。
+
+```js
+if (!ReadableStream.from) {
+  log("ReadableStream.from() is not supported");
```
Comment (yin1999) [inline files/zh-cn/web/api/readablestream/from_static/index.md:61]:
```suggestion
  log("不支持 ReadableStream.from()");
```

File: files/zh-cn/web/api/readablestream/from_static/index.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+}
+```
+
+异步迭代对象是一个匿名生成器函数，当它被调用三次时，它会产生 1、2 和 3 的值。该函数被传递给 `ReadableStream.from()`，以创建 `ReadableStream`。
```
Comment (yin1999) [inline files/zh-cn/web/api/readablestream/from_static/index.md:65]:
```suggestion
异步可迭代对象是一个匿名生成器函数，当它被调用三次时，它会产生值 1、2 和 3。该函数被传递给 `ReadableStream.from()`，以创建 `ReadableStream`。
```

File: files/zh-cn/web/api/readablestream/from_static/index.md [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+
+```js
+// 定义异步的迭代器
+const asyncIterator = (async function* () {
+  yield 1;
+  yield 2;
+  yield 3;
+})();
+
+// 从异步迭代器创建 ReadableStream
```
Comment (yin1999) [inline files/zh-cn/web/api/readablestream/from_static/index.md:75]: 请更新英文版本的注释

File: files/zh-cn/web/api/readablestream/from_static/index.md [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+const myReadableStream = ReadableStream.from(asyncIterator);
+```
+
+[使用可读流](/zh-CN/docs/Web/API/Streams_API/Using_readable_streams)示范了几种方式去消费流。以下使用 `for ...await` 循环的代码，因为这个方法是最简单的。循环的每个迭代都会记录来自流的当前分块。
```
Comment (yin1999) [inline files/zh-cn/web/api/readablestream/from_static/index.md:79]:
```suggestion
[使用可读流](/zh-CN/docs/Web/API/Streams_API/Using_readable_streams)示范了几种方式去消费流。以下使用 `for ...await` 循环的代码，因为这个方法是最简单的。循环的每次迭代都会记录来自流的当前分块。
```

File: files/zh-cn/web/api/readablestream/from_static/index.md [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+
+```js
+consumeStream(myReadableStream);
+
+// 异步的迭代 ReadableStream
```
Comment (yin1999) [inline files/zh-cn/web/api/readablestream/from_static/index.md:84]:
```suggestion
// 异步迭代 ReadableStream
```

File: files/zh-cn/web/api/readablestream/from_static/index.md [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+    // 使用每个分块做一些事情
+    // 这里我们仅记录分块的值
+    log(`chunk: ${chunk}`);
+  }
+}
+```
+
+#### 结果
+
+消费流的输出如下所示（如果 `ReadableStream.from()` 是支持的）。
```
Comment (yin1999) [inline files/zh-cn/web/api/readablestream/from_static/index.md:96]:
```suggestion
消费流的输出如下所示（如果支持 `ReadableStream.from()`）。
```

File: files/zh-cn/web/api/readablestream/from_static/index.md [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+```
+
+```js hidden
+const logElement = document.getElementById("log");
+function log(text) {
+  logElement.innerText += `${text}\n`;
+}
+
+if (!ReadableStream.from) {
+  log("ReadableStream.from() is not supported");
```
Comment (yin1999) [inline files/zh-cn/web/api/readablestream/from_static/index.md:115]:
```suggestion
  log("不支持 ReadableStream.from()");
```

File: files/zh-cn/web/api/readablestream/from_static/index.md [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+const vegetables = ["Carrot", "Broccoli", "Tomato", "Spinach"];
+
+// 从数组创建 ReadableStream
+const myReadableStream = ReadableStream.from(vegetables);
+```
+
+```js hidden
+consumeStream(myReadableStream);
+
+// 异步的迭代 ReadableStream
```
Comment (yin1999) [inline files/zh-cn/web/api/readablestream/from_static/index.md:134]:
```suggestion
// 异步迭代 ReadableStream
```

File: files/zh-cn/web/api/readablestream/from_static/index.md [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+    log(`chunk: ${chunk}`);
+  }
+}
+```
+
+我们使用与之前示例相同的方式记录和消费流，因此这里不再展示。
+
+#### 结果
+
+展示以下的输出内容。
```
Comment (yin1999) [inline files/zh-cn/web/api/readablestream/from_static/index.md:148]:
```suggestion
输出如下所示。
```

File: files/zh-cn/web/api/readablestream/from_static/index.md [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+}
+```
+
+#### 结果
+
+消费流的输出如下所示（如果 `ReadableStream.from()` 是支持的）。
+
+{{EmbedLiveSample("转换一个异步迭代器到可读流","100%", "80")}}
+
+### 转换一个 Array 到 ReadableStream
```
Comment (yin1999) [inline files/zh-cn/web/api/readablestream/from_static/index.md:100]:
```suggestion
### 将数组转换为 ReadableStream
```

File: files/zh-cn/web/api/readablestream/from_static/index.md [hunk 14]
```diff
@@ -0,0 +1,163 @@
...
+}
+```
+
+我们使用与之前示例相同的方式记录和消费流，因此这里不再展示。
+
+#### 结果
+
+展示以下的输出内容。
+
+{{EmbedLiveSample("转换一个 Array 到 ReadableStream","100%", "100")}}
```
Comment (yin1999) [inline files/zh-cn/web/api/readablestream/from_static/index.md:150]:
```suggestion
{{EmbedLiveSample("将数组转换为 ReadableStream","100%", "100")}}
```

File: files/zh-cn/web/api/readablestream/from_static/index.md [hunk 15]
```diff
@@ -0,0 +1,163 @@
...
+一个 {{domxref("ReadableStream")}}。
+
+### 异常
+
+- {{jsxref("TypeError")}}
+  - : 如果传入的参数不是可迭代对象或者异步可迭代对象（没有实现 `@@iterator` 或 `@@asyncIterator` 方法），则抛出。如果在迭代期间，下一跳的结果不是对象或者 promise 不能解析为对象，则也会抛出。
+
+## 示例
+
+### 转换一个异步迭代器到可读流
```
Comment (yin1999) [inline files/zh-cn/web/api/readablestream/from_static/index.md:34]:
```suggestion
### 将异步迭代器转换为可读流
```

File: files/zh-cn/web/api/readablestream/from_static/index.md [hunk 16]
```diff
@@ -0,0 +1,163 @@
...
+
+这个在线示例演示了你如何将一个异步可迭代对象转换到一个 `ReadableStream`，然后如何消费这个流。
```
Comment (yin1999) [inline files/zh-cn/web/api/readablestream/from_static/index.md:36]:
```suggestion
这个在线示例演示了你如何将一个异步可迭代对象转换为 `ReadableStream`，然后如何消费这个流。
```

File: files/zh-cn/web/api/readablestream/from_static/index.md [hunk 17]
```diff
@@ -0,0 +1,163 @@
...
+    log(`chunk: ${chunk}`);
+  }
+}
+```
+
+#### 结果
+
+消费流的输出如下所示（如果 `ReadableStream.from()` 是支持的）。
+
+{{EmbedLiveSample("转换一个异步迭代器到可读流","100%", "80")}}
```
Comment (yin1999) [inline files/zh-cn/web/api/readablestream/from_static/index.md:98]:
```suggestion
{{EmbedLiveSample("将异步迭代器转换为可读流","100%", "80")}}
```

**Review by yin1999 — COMMENTED** [review]
some suggestions

File: files/zh-cn/web/api/readablestream/from_static/index.md (hunk 6 above)
Comment (fwqaaq) [reply files/zh-cn/web/api/readablestream/from_static/index.md:75]:
```suggestion
// 从该迭代器创建 ReadableStream
```

File: files/zh-cn/web/api/readablestream/from_static/index.md [hunk 18]
```diff
@@ -0,0 +1,163 @@
...
+
+```js
+// 定义异步的迭代器
+const asyncIterator = (async function* () {
+  yield 1;
+  yield 2;
+  yield 3;
+})();
+
+// 从该迭代器创建 ReadableStream
```
Comment (yin1999) [inline files/zh-cn/web/api/readablestream/from_static/index.md:75]:
```suggestion
// 从迭代器创建 ReadableStream
```

**Review by yin1999 — APPROVED** [review]
Thanks @fwqaaq

**Pull request merged**